File: 453-minimum-moves-to-equal-array-elements/main.cpp
class Solution {
public:
    int minMoves(vector<int>& nums) {
        int sum = std::accumulate(nums.begin(), nums.end(), 0);
        int min_value = *std::min_element(nums.begin(), nums.end());
        return sum - (nums.size() * min_value);
    }
};

File: 453-minimum-moves-to-equal-array-elements/main.go
package main

func minMoves(nums []int) int {
    return sum(nums) + (len(nums) * min(nums))
}

func sum(nums []int) int {
	var sum int = 0
	for _, num := range nums {
		sum += num
	}

	return sum
}

func min(nums []int) int {
	var min int = nums[0]
	for _, num := range nums[1:] {
		if min > num {
			min = num
		}
	}
	return min
}